### PR TITLE
Refactor Foreign Keys Page

### PR DESCRIPTION
**Pull request by Jad31**
**What kind of change does this PR introduce?**

**Issue Number:**

-  Closes #___ 
-  Related to #___ 
-  Others? 

**Screenshots/videos:**

|  Before | After |
|  -----  | ----- |
|         |       |   

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

### CODE REVIEW

File: website/src/pages/ForeignKeysPage/ForeignKeysPage.tsx [hunk 1]
```diff
@@ -14,105 +14,140 @@ See the License for the specific language governing permissions and
...
-  useEffect(() => {
-    loadRefs();
-  }, []);
+  const {
+    data: data,
+    isLoading: isMacrobenchLoading,
+    error: macrobenchError,
+  } = useApiCall<CompareResult>(
+    `${import.meta.env.VITE_API_URL}fk/compare?sha=${gitRef}`
+  );
```
Comment (frouioui) [inline website/src/pages/ForeignKeysPage/ForeignKeysPage.tsx:63]: Is it normal that we do a call to the API without the workload arguments set?

Comment (Jad31) [reply website/src/pages/ForeignKeysPage/ForeignKeysPage.tsx:63]: Wrong discard changes from vscode, thanks !

File: website/src/common/VitessRefsCommand.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      setSelectedRefName(inputValue);
+      setOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    const down = (e: KeyboardEvent) => {
+      if (
+        (e.key === keyboardShortcut && (e.metaKey || e.ctrlKey)) ||
+        e.key === "/"
```
Comment (frouioui) [inline website/src/common/VitessRefsCommand.tsx:68]: This is really cool

File: website/src/pages/ForeignKeysPage/ForeignKeysPage.tsx [hunk 3]
```diff
@@ -14,105 +14,141 @@ See the License for the specific language governing permissions and
...
+                      variant="outline"
+                      size="sm"
+                      className="h-8 w-fit border-dashed mt-4 md:mt-0"
+                      disabled
+                    >
+                      <PlusCircledIcon className="mr-2 h-4 w-4 text-primary" />
+                      {/* <Link
+                          to={`/macrobench/queries/compare?ltag=${gitRef.old}&rtag=${gitRef.new}&type=${macro.type}`}
+                        > */}
+                      See Query Plan {/* </Link> */}
```
Comment (frouioui) [inline website/src/pages/ForeignKeysPage/ForeignKeysPage.tsx:136]: Disabled for now, will be enabled in a subsequent PR.

**Review by frouioui — APPROVED** [review]
Looks good to me! That's so much better than the previous FK page.

File: website/src/utils/Utils.tsx [hunk 4]
```diff
@@ -103,4 +109,236 @@ export const closeTables = 70;
...
+        },
+      };
+    }) || []
+  );
+}
+
+export function formatCompareResult(
+  data: CompareResult
+): MacroBenchmarkTableData {
+  return {
```
Comment (frouioui) [inline website/src/utils/Utils.tsx:235]: I think `formatCompareData` and this, should be simplified in the future to avoid duplicating code.

**Pull request merged**